headerfs/truncate.go

```diff
@@ -22,18 +22,27 @@ func (h *headerStore) singleTruncate() error {
 
 	// Next, we'll determine the number of bytes we need to truncate from
 	// the end of the file.
-	var truncateLength int64
-	switch h.indexType {
-	case Block:
-		truncateLength = 80
-	case RegularFilter:
-		truncateLength = 32
-	default:
-		return fmt.Errorf("unknown index type: %v", h.indexType)
+	truncateLength, err := h.indexType.Size()
+	if err != nil {
+		return err
+	}
+	newSize := fileSize - int64(truncateLength)
+
+	// On Windows, we need to close, truncate, and reopen the file.
+	if runtime.GOOS == "windows" {
```

Comment:
Not bad, saves some extra files, while also putting more of the logic in line vs normal build tags.